### ISSUE TITLE
snapshot works locally, but not on CircleCI

### ISSUE THREAD

**Issue by chiformihai** [fastlane/fastlane#17193]
### New Issue Checklist

- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] Updated fastlane to the latest version
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Hello, I'm trying to automate the screenshots generation for an iOS app on CircleCI and I'm getting a strange error right after launching the simulator and before starting to build the app for testing. 
Executing the exact same lane locally works without a problem and the screenshots are generated as expected.

> bundler: failed to load command: fastlane (/Users/distiller/rubik/vendor/bundle/ruby/2.7.0/bin/fastlane)
NoMethodError: [!] undefined method `map' for "devices":String

It looks like at some point it tries to "map" over "devices" which instead of being an array, is a String.

##### Command executed

`bundle exec fastlane ios generate_screenshots`

where **generate_screenshots** is

`lane :generate_screenshots do |options|
    pre_build_configuration(options)
    devices = ["iPhone Test"]
    snapshot(
        reinstall_app: true,
        app_identifier: bundleId,
        devices: devices,
        scheme: ...,
        workspace: ...,
        configuration: "Release",
        languages: [
          "en-US"
        ]
    )
  end`

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>
bundler: failed to load command: fastlane (/Users/distiller/rubik/vendor/bundle/ruby/2.7.0/bin/fastlane)
NoMethodError: [!] undefined method `map' for "devices":String
Did you mean?  tap
Did you mean?  tap
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/simctl-1.6.8/lib/simctl/command/list.rb:27:in `block in list_devices'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/simctl-1.6.8/lib/simctl/executor.rb:20:in `block in execute'
  /Users/distiller/.rubies/ruby-2.7.1/lib/ruby/2.7.0/open3.rb:219:in `popen_run'
  /Users/distiller/.rubies/ruby-2.7.1/lib/ruby/2.7.0/open3.rb:101:in `popen3'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/simctl-1.6.8/lib/simctl/executor.rb:10:in `execute'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/simctl-1.6.8/lib/simctl/command/list.rb:26:in `list_devices'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/simctl-1.6.8/lib/simctl/command/list.rb:9:in `device'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/simctl-1.6.8/lib/simctl.rb:39:in `method_missing'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/test_command_generator.rb:100:in `block in get_device_type_with_simctl'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/test_command_generator.rb:99:in `map'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/test_command_generator.rb:99:in `get_device_type_with_simctl'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/test_command_generator.rb:67:in `verify_devices_share_os'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/test_command_generator.rb:42:in `destination'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/test_command_generator.rb:15:in `generate'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:85:in `launch_simultaneously'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:70:in `block (3 levels) in take_screenshots_simultaneously'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:69:in `each'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:69:in `block (2 levels) in take_screenshots_simultaneously'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:46:in `each'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:46:in `each_with_index'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:46:in `block in take_screenshots_simultaneously'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:45:in `each'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/simulator_launchers/simulator_launcher.rb:45:in `take_screenshots_simultaneously'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/snapshot/lib/snapshot/runner.rb:45:in `work'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/actions/capture_ios_screenshots.rb:14:in `run'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:229:in `chdir'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
  Fastfile:473:in `block (2 levels) in parsing_binding'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/gems/fastlane-2.158.0/bin/fastlane:23:in `<top (required)>'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/bin/fastlane:23:in `load'
  /Users/distiller/rubik/vendor/bundle/ruby/2.7.0/bin/fastlane:23:in `<top (required)>'
</pre>
</details>

### Circle CI Environment

v10.15.1 is already installed.
Now using node v10.15.1 (npm v6.4.1)
default -> 10.15.1 (-> v10.15.1 *)
[⠋] 🚀 [⠙] 🚀 [⠹] 🚀 [⠸] 🚀 [⠼] 🚀 [⠴] 🚀 [⠦] 🚀 [⠧] 🚀 [⠇] 🚀 [⠏] 🚀 [✔] 🚀 
[10:44:23]: Sending anonymous analytics information
[10:44:23]: Learn more at https://docs.fastlane.tools/#metrics
[10:44:23]: No personal or sensitive data is sent.
[10:44:23]: You can disable this by adding `opt_out_usage` at the top of your Fastfile
[10:44:23]: Generating fastlane environment output, this might take a few seconds...
<details><summary>✅ fastlane environment from CircleCI ✅</summary>

### Stack

| Key                         | Value                                            |
| --------------------------- | ------------------------------------------------ |
| OS                          | 10.15.5                                          |
| Ruby                        | 2.7.1                                            |
| Bundler?                    | true                                             |
| Git                         | git version 2.28.0                               |
| Installation Source         | ~/rubik/vendor/bundle/ruby/2.7.0/bin/fastlane    |
| Host                        | Mac OS X 10.15.5 (19F101)                        |
| Ruby Lib Dir                | ~/.rubies/ruby-2.7.1/lib                         |
| OpenSSL Version             | OpenSSL 1.1.1g  21 Apr 2020                      |
| Is contained                | false                                            |
| Is homebrew                 | false                                            |
| Is installed via Fabric.app | false                                            |
| Xcode Path                  | /Applications/Xcode-11.7.app/Contents/Developer/ |
| Xcode Version               | 11.7                                             |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_require 'dotenv'
fastlane_require 'spaceship'
fastlane_require 'xcodeproj'
fastlane_require 'plist'
release_notes_command = "git log HEAD --pretty=format:\"%s\" -1"

...

# IOS

platform :ios do |options|

  lane :generate_screenshots do |options|
    pre_build_configuration(options)

    scheme = "..."
    workspace = "..."
    devices = ["iPhone Test"]

    snapshot(
        reinstall_app: true,
        app_identifier: bundleId,
        devices: devices,
        scheme: scheme,
        workspace: workspace,
        configuration: "Release",
        languages: [
          "en-US"
        ]
    )
  end

  ...
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# See https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
app_identifier ENV['IOS_APP_ID']
apple_id ENV['IOS_USER_ID']
team_id ENV['IOS_TEAM_ID']
itc_team_name ENV['IOS_ITC_TEAM_NAME']

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.158.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                                | Version | Update-Status |
| ----------------------------------------------------- | ------- | ------------- |
| fastlane-plugin-appcenter                             | 1.9.0   | ✅ Up-To-Date  |
| fastlane-plugin-update_provisioning_profile_specifier | 1.3.1   | ✅ Up-To-Date  |
| fastlane-plugin-load_json                             | 0.0.1   | ✅ Up-To-Date  |
| fastlane-plugin-aws_s3                                | 1.8.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                                   | Version      |
| ----------------------------------------------------- | ------------ |
| did_you_mean                                          | 1.4.0        |
| bundler                                               | 2.1.4        |
| uri                                                   | 0.10.0       |
| rake                                                  | 13.0.1       |
| CFPropertyList                                        | 3.0.2        |
| concurrent-ruby                                       | 1.1.7        |
| i18n                                                  | 0.9.5        |
| minitest                                              | 5.14.2       |
| thread_safe                                           | 0.3.6        |
| tzinfo                                                | 1.2.7        |
| activesupport                                         | 4.2.11.3     |
| public_suffix                                         | 4.0.6        |
| addressable                                           | 2.7.0        |
| httpclient                                            | 2.8.3        |
| json                                                  | 2.3.1        |
| algoliasearch                                         | 1.27.3       |
| rubyzip                                               | 2.3.0        |
| apktools                                              | 0.7.4        |
| atomos                                                | 0.1.3        |
| aws-eventstream                                       | 1.1.0        |
| aws-partitions                                        | 1.365.0      |
| aws-sigv4                                             | 1.2.2        |
| jmespath                                              | 1.4.0        |
| aws-sdk-core                                          | 3.105.0      |
| aws-sdk-kms                                           | 1.37.0       |
| aws-sdk-s3                                            | 1.79.1       |
| babosa                                                | 1.0.3        |
| claide                                                | 1.0.3        |
| fuzzy_match                                           | 2.0.4        |
| nap                                                   | 1.1.0        |
| netrc                                                 | 0.11.0       |
| ffi                                                   | 1.13.1       |
| ethon                                                 | 0.12.0       |
| typhoeus                                              | 1.4.0        |
| cocoapods-core                                        | 1.9.3        |
| cocoapods-deintegrate                                 | 1.0.4        |
| cocoapods-downloader                                  | 1.4.0        |
| cocoapods-plugins                                     | 1.0.0        |
| cocoapods-search                                      | 1.0.0        |
| cocoapods-stats                                       | 1.1.0        |
| cocoapods-trunk                                       | 1.5.0        |
| cocoapods-try                                         | 1.2.0        |
| colored2                                              | 3.1.2        |
| escape                                                | 0.0.4        |
| fourflusher                                           | 2.3.1        |
| gh_inspector                                          | 1.1.3        |
| molinillo                                             | 0.6.6        |
| ruby-macho                                            | 1.4.0        |
| nanaimo                                               | 0.3.0        |
| xcodeproj                                             | 1.18.0       |
| cocoapods                                             | 1.9.3        |
| colored                                               | 1.2          |
| highline                                              | 1.7.10       |
| commander-fastlane                                    | 4.4.6        |
| declarative                                           | 0.0.20       |
| declarative-option                                    | 0.1.0        |
| digest-crc                                            | 0.6.1        |
| unf_ext                                               | 0.0.7.7      |
| unf                                                   | 0.1.4        |
| domain_name                                           | 0.5.20190701 |
| dotenv                                                | 2.7.6        |
| emoji_regex                                           | 3.0.0        |
| excon                                                 | 0.76.0       |
| multipart-post                                        | 2.0.0        |
| faraday                                               | 1.0.1        |
| http-cookie                                           | 1.0.3        |
| faraday-cookie_jar                                    | 0.0.7        |
| faraday_middleware                                    | 1.0.0        |
| fastimage                                             | 2.2.0        |
| jwt                                                   | 2.2.2        |
| memoist                                               | 0.16.2       |
| multi_json                                            | 1.15.0       |
| os                                                    | 1.1.1        |
| signet                                                | 0.14.0       |
| googleauth                                            | 0.13.1       |
| mini_mime                                             | 1.0.2        |
| uber                                                  | 0.1.0        |
| representable                                         | 3.0.4        |
| retriable                                             | 3.1.2        |
| google-api-client                                     | 0.38.0       |
| google-cloud-env                                      | 1.3.3        |
| google-cloud-errors                                   | 1.0.1        |
| google-cloud-core                                     | 1.5.0        |
| google-cloud-storage                                  | 1.28.0       |
| mini_magick                                           | 4.10.1       |
| plist                                                 | 3.5.0        |
| security                                              | 0.1.3        |
| naturally                                             | 2.2.0        |
| simctl                                                | 1.6.8        |
| slack-notifier                                        | 2.3.2        |
| terminal-notifier                                     | 2.0.0        |
| unicode-display_width                                 | 1.7.0        |
| terminal-table                                        | 1.8.0        |
| tty-screen                                            | 0.8.1        |
| tty-cursor                                            | 0.7.1        |
| tty-spinner                                           | 0.9.3        |
| word_wrap                                             | 1.0.0        |
| rouge                                                 | 2.0.7        |
| xcpretty                                              | 0.3.0        |
| xcpretty-travis-formatter                             | 1.0.0        |
| fastlane-plugin-appcenter                             | 1.9.0        |
| mime-types-data                                       | 3.2020.0512  |
| mime-types                                            | 3.3.1        |
| fastlane-plugin-aws_s3                                | 1.8.0        |
| fastlane-plugin-load_json                             | 0.0.1        |
| fastlane-plugin-update_provisioning_profile_specifier | 1.3.1        |
| mini_portile2                                         | 2.4.0        |
| nokogiri                                              | 1.10.10      |
</details>
</details>

### Local environment
<details><summary>🚫 Local fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                         |
| --------------------------- | ------------------------------------------------------------- |
| OS                          | 10.15.5                                                       |
| Ruby                        | 2.6.0                                                         |
| Bundler?                    | true                                                          |
| Git                         | git version 2.24.3 (Apple Git-128)                            |
| Installation Source         | ~/projects/rubik-mobile/vendor/bundle/ruby/2.6.0/bin/fastlane |
| Host                        | Mac OS X 10.15.5 (19F101)                                     |
| Ruby Lib Dir                | ~/.rbenv/versions/2.6.0/lib                                   |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                                   |
| Is contained                | false                                                         |
| Is homebrew                 | false                                                         |
| Is installed via Fabric.app | false                                                         |
| Xcode Path                  | /Applications/Xcode-beta.app/Contents/Developer/              |
| Xcode Version               | 12.0                                                          |

### System Locale

| Variable | Value             |   |
| -------- | ----------------- | - |
| LANG     | en_US.UTF-8export | 🚫 |
| LC_ALL   | en_US.UTF-8       | ✅ |
| LANGUAGE |                   |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_require 'dotenv'
fastlane_require 'spaceship'
fastlane_require 'xcodeproj'
fastlane_require 'plist'
release_notes_command = "git log HEAD --pretty=format:\"%s\" -1"

...

# IOS

platform :ios do |options|

  lane :generate_screenshots do |options|
    pre_build_configuration(options)

    scheme = "..."
    workspace = "..."
    devices = ["iPhone Test"]

    snapshot(
        reinstall_app: true,
        app_identifier: bundleId,
        devices: devices,
        scheme: scheme,
        workspace: workspace,
        configuration: "Release",
        languages: [
          "en-US"
        ]
    )
  end

  ...
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# See https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md
app_identifier ENV['IOS_APP_ID']
apple_id ENV['IOS_USER_ID']
team_id ENV['IOS_TEAM_ID']
itc_team_name ENV['IOS_ITC_TEAM_NAME']

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.158.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                                | Version | Update-Status |
| ----------------------------------------------------- | ------- | ------------- |
| fastlane-plugin-appcenter                             | 1.9.0   | ✅ Up-To-Date  |
| fastlane-plugin-update_provisioning_profile_specifier | 1.3.1   | ✅ Up-To-Date  |
| fastlane-plugin-load_json                             | 0.0.1   | ✅ Up-To-Date  |
| fastlane-plugin-aws_s3                                | 1.8.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                                   | Version      |
| ----------------------------------------------------- | ------------ |
| did_you_mean                                          | 1.3.0        |
| bundler                                               | 2.1.4        |
| rake                                                  | 13.0.1       |
| CFPropertyList                                        | 3.0.2        |
| concurrent-ruby                                       | 1.1.7        |
| i18n                                                  | 0.9.5        |
| minitest                                              | 5.14.2       |
| thread_safe                                           | 0.3.6        |
| tzinfo                                                | 1.2.7        |
| activesupport                                         | 4.2.11.3     |
| public_suffix                                         | 4.0.6        |
| addressable                                           | 2.7.0        |
| httpclient                                            | 2.8.3        |
| json                                                  | 2.3.1        |
| algoliasearch                                         | 1.27.3       |
| rubyzip                                               | 2.3.0        |
| apktools                                              | 0.7.4        |
| atomos                                                | 0.1.3        |
| aws-eventstream                                       | 1.1.0        |
| aws-partitions                                        | 1.365.0      |
| aws-sigv4                                             | 1.2.2        |
| jmespath                                              | 1.4.0        |
| aws-sdk-core                                          | 3.105.0      |
| aws-sdk-kms                                           | 1.37.0       |
| aws-sdk-s3                                            | 1.79.1       |
| babosa                                                | 1.0.3        |
| claide                                                | 1.0.3        |
| fuzzy_match                                           | 2.0.4        |
| nap                                                   | 1.1.0        |
| netrc                                                 | 0.11.0       |
| ffi                                                   | 1.13.1       |
| ethon                                                 | 0.12.0       |
| typhoeus                                              | 1.4.0        |
| cocoapods-core                                        | 1.9.0        |
| cocoapods-deintegrate                                 | 1.0.4        |
| cocoapods-downloader                                  | 1.4.0        |
| cocoapods-plugins                                     | 1.0.0        |
| cocoapods-search                                      | 1.0.0        |
| cocoapods-stats                                       | 1.1.0        |
| cocoapods-trunk                                       | 1.5.0        |
| cocoapods-try                                         | 1.2.0        |
| colored2                                              | 3.1.2        |
| escape                                                | 0.0.4        |
| fourflusher                                           | 2.3.1        |
| gh_inspector                                          | 1.1.3        |
| molinillo                                             | 0.6.6        |
| ruby-macho                                            | 1.4.0        |
| nanaimo                                               | 0.3.0        |
| xcodeproj                                             | 1.18.0       |
| cocoapods                                             | 1.9.0        |
| colored                                               | 1.2          |
| highline                                              | 1.7.10       |
| commander-fastlane                                    | 4.4.6        |
| declarative                                           | 0.0.20       |
| declarative-option                                    | 0.1.0        |
| digest-crc                                            | 0.6.1        |
| unf_ext                                               | 0.0.7.7      |
| unf                                                   | 0.1.4        |
| domain_name                                           | 0.5.20190701 |
| dotenv                                                | 2.7.6        |
| emoji_regex                                           | 3.0.0        |
| excon                                                 | 0.76.0       |
| multipart-post                                        | 2.0.0        |
| faraday                                               | 1.0.1        |
| http-cookie                                           | 1.0.3        |
| faraday-cookie_jar                                    | 0.0.7        |
| faraday_middleware                                    | 1.0.0        |
| fastimage                                             | 2.2.0        |
| jwt                                                   | 2.2.2        |
| memoist                                               | 0.16.2       |
| multi_json                                            | 1.15.0       |
| os                                                    | 1.1.1        |
| signet                                                | 0.14.0       |
| googleauth                                            | 0.13.1       |
| mini_mime                                             | 1.0.2        |
| uber                                                  | 0.1.0        |
| representable                                         | 3.0.4        |
| retriable                                             | 3.1.2        |
| google-api-client                                     | 0.38.0       |
| google-cloud-env                                      | 1.3.3        |
| google-cloud-errors                                   | 1.0.1        |
| google-cloud-core                                     | 1.5.0        |
| google-cloud-storage                                  | 1.28.0       |
| mini_magick                                           | 4.10.1       |
| plist                                                 | 3.5.0        |
| security                                              | 0.1.3        |
| naturally                                             | 2.2.0        |
| simctl                                                | 1.6.8        |
| slack-notifier                                        | 2.3.2        |
| terminal-notifier                                     | 2.0.0        |
| unicode-display_width                                 | 1.7.0        |
| terminal-table                                        | 1.8.0        |
| tty-screen                                            | 0.8.1        |
| tty-cursor                                            | 0.7.1        |
| tty-spinner                                           | 0.9.3        |
| word_wrap                                             | 1.0.0        |
| rouge                                                 | 2.0.7        |
| xcpretty                                              | 0.3.0        |
| xcpretty-travis-formatter                             | 1.0.0        |
| fastlane-plugin-appcenter                             | 1.9.0        |
| mime-types-data                                       | 3.2020.0512  |
| mime-types                                            | 3.3.1        |
| fastlane-plugin-aws_s3                                | 1.8.0        |
| fastlane-plugin-load_json                             | 0.0.1        |
| fastlane-plugin-update_provisioning_profile_specifier | 1.3.1        |
| mini_portile2                                         | 2.4.0        |
| nokogiri                                              | 1.10.10      |
</details>

*generated on:* **2020-09-09**
</details>
</details>

As you can see the environments are a little bit different, but I tried even with the exact configuration that's currently on Circle and it still produces the exact same result.

"iPhone Test" is a device that I'm creating and booting before executing the **generate_screenshots** command

Any help is greatly appreciated. Thanks!

**Comment by tmaly1980** [comment]
From what it sounds like, it's almost as if your devices parameter to snapshot is a string, rather than an array. Even though I see how you assign it as an array prior. Have you tried running "UI.message devices" (prior to calling snapshot, to confirm it's an array) or hardcoding the device list inside snapshot()?

**Comment by chiformihai** [comment]
I am currently passing the devices to snapshot, but I also tried specifying them in Snapfile and a combination of both and still get the same error.

I also tried executing the action using `bundle exec fastlane snapshot --devices [] ...` and then passing the parameters there, with the same result.

I saw that fastlane is using Simctl for booting up devices and I used it too to list all the devices and everything seemed to be fine. 

I tried resetting all the simulators before executing snapshot, I tried deleting all the simulators to make sure I only have 1 simulator installed and booted up, still didn't help (did all these actions using `xcrun simctl ...`)

**Comment by chiformihai** [comment]
One other thing that I tried was to run **scan** with _derived_data_path_  and _build_for_testing: true_ and then to execute **snapshot** with _test_without_building: true_ but with this method it fails @ **scan** with 

> Malformed version number string unknown
error

Again, this method worked perfectly fine on my local machine

**Comment by tmaly1980** [comment]
This is basically what I'm running:

```
          snapshot(
            app_identifier: 'com.company.appid',
            output_directory: "fastlane/metadata/en-US/images/screenshots",
            devices: ['iPhone 8 Plus'],
            derived_data_path: './DerivedData',
            languages: ['en-US'],
            project: './AppName.xcodeproj',
            skip_open_summary: true,
            number_of_retries: 0,
            reinstall_app: true,
            clear_previous_screenshots: true,
          )
```

I don't follow your last comment, can you show specifics (ie code)?

**Comment by chiformihai** [comment]
`scheme = "ScreenshotsScheme"
    workspace = "./ios/app.xcworkspace"
    path = "~/Library/Developer/Xcode/DerivedData/AppDevelopment"
    devices = [
      "iPhone 11"
    ]

    scan(
        build_for_testing: true,
        code_coverage: false,
        derived_data_path: path,
        devices: devices,
        scheme: scheme,
        workspace: workspace,
        reset_simulator: true,
        prelaunch_simulator: true,
        app_identifier: bundleId,
        configuration: "Release"
    )

    snapshot(
        reinstall_app: true,
        app_identifier: bundleId,
        devices: devices,
        scheme: scheme,
        workspace: workspace,
        configuration: "Release",
        derived_data_path: path,
        languages: ["en-US"],
        test_without_building: true
    )`

The idea was to build the app for testing and then use the build to generate the screenshots

**Comment by chmiiller** [comment]
Do you think that https://github.com/fastlane/fastlane/issues/12986 is related? Someone is mentioning there that: 

> it seems like this might be a race condition from within the simctl gem that we use to help us with some simulator querying. 

Don't know, just trying to help 😬

**Comment by chiformihai** [comment]
I just got it working with a different XCode image set up on Circle. I tried only with the most recent ones, but **11.2.1** seems to be the winner.

We're limited on using node version 10 or 12 due to some dependencies in our project, so this Xcode image is the most recent that comes with node 12, for the other ones I had to manually install node 10 during Circle setup and maybe I was messing something up there

This is the command that I was using for that 
`- run:
          name: "install node@10.15.1"
          command: |
            set +e
            touch $BASH_ENV
            curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.33.5/install.sh | bash
            echo 'export NVM_DIR="$HOME/.nvm"' >> $BASH_ENV
            echo '[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"' >> $BASH_ENV
            echo nvm install 10.15.1 >> $BASH_ENV
            echo nvm alias default 10.15.1 >> $BASH_ENV`

It's still not clear to me why does this affects fastlane

**Comment by Edwar1d** [comment]
Thanks for the update and quick reply. I'll be sure to keep an eye on this thread [tellthebell](https://tellthebell.ltd/).
